runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/DoFnOp.java

```diff
@@ -330,9 +333,12 @@ private void doProcessWatermark(Instant watermark, OpEmitter<OutT> emitter) {
     timerInternalsFactory.setInputWatermark(actualInputWatermark);
 
     Collection<? extends KeyedTimerData<?>> readyTimers = timerInternalsFactory.removeReadyTimers();
-    if (!readyTimers.isEmpty()) {
+    // Process timers in chunks
+    Iterator<? extends KeyedTimerData<?>> timerIterator = readyTimers.iterator();
+    while (timerIterator.hasNext()) {
```

Comment:
I think this might not work as expected. Even the timeres are processed in chunks, the output will be collected for all the trunks, since they are processed in a single Samza flatmap function. Seems to me we have to break this DoFnOp into two steps, first is to emit the timers in batches, and then the second is to process the timers in batches.